pages/routerGuard.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import { getSession, signin } from 'next-auth/client';
+import { useRouter } from 'next/router';
+import { DateTime } from 'luxon';
+
+export const RouterGuard: React.FC = ({ children }) => {
+  const [isAuthed, setIsAuthed] = useState(false);
```

Comment:
Is there any way to get the status `isAuthed` from a hook or an api directly without using `useState`?

Reply:
There is a ```useSession``` hook and I attempted a very similar setup. However, it didn't work as expected. ```useSession``` grabs the context from the session ```Provider``` in ```_app.page.tsx```,  while ```getSession``` is making an api call to ```/api/auth/session``` to get the most recent and updated state of the session. I noticed if I deleted my session from cookies for example and then navigated to another page, ```useSession``` would still return that I have a valid session, while ```getSession``` would have the correct and expected value.

Reply:
So essentially you're wanting to update the session on every page navigation? That fires off a network request to the Next.js backend every time. I guess that's ok... Seems like a little overkill maybe. I think the initial app load will send this request twice, once in `SessionProvider` and again immediately here. We could wait until a GraphQL request fails, which probably happens around the same time since most pages fire off a query during render.

[The docs](https://next-auth.js.org/getting-started/client#options) say
>  If you need to, you can trigger an update of the session object across all tabs/windows by calling getSession() from a client side function.

That sounds like you may not need to store anything locally. My assumption is `getSession` should make updates that `useSession` picks up which would rerender the component. Maybe it doesn't work that way but that sounds cleaner.

[`refetchInterval`](https://next-auth.js.org/getting-started/client#options) is also an option. Idk that we have any specific need for it but might be a cleaner way to accomplish session updating.